tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -629,67 +572,44 @@ describe(__filename, () => {
   });
 
   it('does not render an edit link if no user found', () => {
-    const root = renderUserProfile({ params: { username: 'not-loaded' } });
+    const root = renderUserProfile({ params: { userId: 1234 } });
 
     expect(root.find('.UserProfile-edit-link')).toHaveLength(0);
   });
 
   it('renders an edit link if user has sufficient permission', () => {
-    const username = 'current-logged-in-user';
-    const { store } = dispatchSignInActions({
-      userProps: {
-        username,
-        permissions: [USERS_EDIT],
-      },
+    const { store } = signInUserWithProps({
+      userId: 1,
+      permissions: [USERS_EDIT],
     });
 
-    // Create a user with another username.
-    const user = createUserAccountResponse({ username: 'willdurand' });
+    // Create a user with another userId.
+    const user = createUserAccountResponse({ id: 2 });
     store.dispatch(loadUserAccount({ user }));
 
     // See this other user profile page.
-    const params = { username: user.username };
+    const params = { userId: user.id };
     const root = renderUserProfile({ params, store });
 
     expect(root.find('.UserProfile-edit-link')).toHaveLength(1);
     expect(root.find('.UserProfile-edit-link')).toHaveProp(
       'to',
-      `/user/${user.username}/edit/`,
+      `/user/${user.id}/edit/`,
     );
   });
 
   it('does not render an edit link if user is not allowed to edit other users', () => {
-    const username = 'current-logged-in-user';
-    const { store } = dispatchSignInActions({
-      userProps: {
-        username,
-        permissions: [],
-      },
+    const { store } = signInUserWithProps({
+      userId: 1,
+      permissions: [],
     });
 
-    // Create a user with another username.
-    const user = createUserAccountResponse({ username: 'willdurand' });
+    // Create a user with another userId.
+    const user = createUserAccountResponse({ id: 2 });
     store.dispatch(loadUserAccount({ user }));
 
     // See this other user profile page.
-    const params = { username: user.username };
-    const root = renderUserProfile({ params, store });
-
-    expect(root.find('.UserProfile-edit-link')).toHaveLength(0);
-  });
-
-  it('does not render an edit link if user is not found', () => {
```

Comment:
Why is this test no longer needed?

Reply:
It is already present, no? https://github.com/mozilla/addons-frontend/blob/ae8f4eccfe67520097495d174750d0d9084a4389/tests/unit/amo/pages/TestUserProfile.js#L574-L578